### DISCUSSION TITLE
FUSE mount consistency?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [seaweedfs/seaweedfs#5904] [Q&A]
If multiple non-cluster clients have a directory mounted, are file operations consistent? (at least [open-to-close consistency](https://juicefs.com/docs/community/getting-started/for_distributed/#strong-data-consistency-guarantee))

I see that according to the [wiki](https://github.com/seaweedfs/seaweedfs/wiki/FUSE-Mount#weed-mount-performance)

> asynchronously replicate the metadata updates to local db. There are no remote metadata read operations at all.

Is this strictly true, that it will never read remote metadata? I would expect that if it does not find file metadata locally, it will attempt to read it from the filter server. And if that's the case, does it become consistent?

To be extra clear, what I am hoping for is that when a remote non-cluster client writes a file, the local non-cluster client would see that subsequent write instantly for a read or file listing.

**Comment by chrislusf** [comment]
I only see "close to open" consistency mentioned in the link, which is basically the same.

What is "instantly"? There could be some milliseconds delay for subscribed metadata messages to arrive in the mount.

Seems you can easily test it out for your use case.

**Reply by danthegoodman1** [reply]
Sorry that’s what I meant XD. So it’s not consistent? Eg it will not check the filer if not in the local metadata? The use case would be shared access to files where it’s expected that the file is available to peers once it is written.

I don’t think it’s easy to test because it’s hard to simulate bad network conditions for code I’m not writing (metadata sync)

**Reply by danthegoodman1** [reply]
To clarify what my use case is, I’m interested in testing rocksdb and badgerdb on top of seaweedfs mounts as a bottomless volume.

the thing that would need to be guaranteed is that when a storage process moves hosts (eg host died, decommissioned) that the new host can guarantee it will have an up to date view of the lsm. If it’s not consistent, then it’s not guaranteed.

also I wonder if there will be issues with the WAL performance

**Reply by danthegoodman1** [reply]
Or maybe is there a way to check/wait for it to sync?

**Reply by chrislusf** [reply]
so you just need close-to-open consistency. it should work.

**Reply by danthegoodman1** [reply]
Right, but you said that it has to wait for metadata to be sent to another mount. So in the event of network issues and that metadata never arrives, a different mount that takes over could be operating on incomplete data because the mount does to fallback to the filer when the file doesn’t exist in the local metadata, correct?

**Reply by chrislusf** [reply]
you can start the other mount after the first mount stops. So all the metadata would come from the filer.

**Reply by danthegoodman1** [reply]
But this relies on me making the mount after, yes? I would imagine that I want to pre-mount when possible so I’d ideally already have the next machine mounted and ready to go

**Reply by chrislusf** [reply]
in general, do not put the db on it. the WAL needs to be flushed before crashing to avoid data loss.

**Reply by danthegoodman1** [reply]
Well it’s easy to configure fsync for rocksdb and others, but I’d imagine this wasn’t a great fit, just wanted to check
